qualtran/drawing/qpic_diagram.py

```diff
@@ -0,0 +1,332 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Classes for drawing latex diagrams for bloqs with QPIC - https://github.com/qpic/qpic.
+
+QPIC is not a dependency of Qualtran and must be manually installed by users via
+`pip install qpic`.
+"""
+import os
+import pathlib
+import re
+import shutil
+import subprocess
+import tempfile
+from collections import defaultdict
+from typing import Dict, List, Optional, Set, Tuple, TYPE_CHECKING, Union
+
+from qualtran import DanglingT, LeftDangle, QBit, Side, Soquet
+from qualtran.drawing.musical_score import _soq_to_symb, Circle, ModPlus, WireSymbol
+
+if TYPE_CHECKING:
+    from qualtran import Bloq, Connection, Signature
+
+
+def _wire_name_prefix_for_soq(soq: Soquet) -> str:
+    return soq.pretty()
+
+
+def _format_label_text(label: str, scale: Optional[float] = None) -> str:
+    replacements = {'&': r'\&', '_': r'\_', '<': r'$<$', '>': r'$>$', '⨁': r'$\oplus$'}
+    for key, val in replacements.items():
+        label = label.replace(key, val)
+    scale_prefix = r'\scalebox{' + str(scale) + r'}{' if scale else ''
+    scale_suffix = r'}' if scale else ''
+    return r'\textrm{' + scale_prefix + str(label) + scale_suffix + r'}'
```

Comment:
you call `str` on label but are already calling `.replace` on it, so this duck typing won't add anything

Reply:
removed